rotkehlchen/chain/ethereum/modules/curve/curve_cache.py

```diff
@@ -215,172 +254,190 @@ def save_curve_data_to_cache(
+    address_provider = ethereum.contracts.contract('CURVE_ADDRESS_PROVIDER')
+    metaregistry_address = to_checksum_address(address_provider.call(
         node_inquirer=ethereum,
         method_name='get_address',
-        arguments=[3],  # 3 is metapool factory id
-    )
-    factory_address = to_checksum_address(factory_address_result)
-    factory_contract = EvmContract(
-        address=factory_address,
-        abi=ethereum.contracts.abi('CURVE_METAPOOL_FACTORY'),
+        arguments=[7],
+    ))
+    metaregistry = EvmContract(
+        address=metaregistry_address,
+        abi=ethereum.contracts.abi('CURVE_METAREGISTRY'),  # type: ignore[call-overload]  # for some reason mypy doesn't properly see argument type  # noqa: E501
         deployed_block=0,  # deployment_block is not used and the contract is dynamic
     )
-    pool_count = factory_contract.call(
-        node_inquirer=ethereum,
-        method_name='pool_count',
-    )
-    pool_addrs_result = ethereum.multicall_specific(
-        contract=factory_contract,
-        method_name='pool_list',
-        arguments=[(x,) for x in range(pool_count)],
-        decode_result=False,  # don't decode since decoding unchecksums address
-    )
-    pool_addresses = []
-    for encoded_pool_addr in pool_addrs_result:
-        pool_addresses.append(hex_or_bytes_to_address(encoded_pool_addr))
-
-    # either a pool of stablecoins or a pair `stablecoin - other curve pool token`
-    # so no need to query underlying tokens
-    pool_coins_result = ethereum.multicall_specific(
-        contract=factory_contract,
-        method_name='get_coins',
-        arguments=[(x,) for x in pool_addresses],
-    )
-    pools_mapping: CURVE_POOLS_MAPPING_TYPE = {}
-    for pool_addr, (decoded_coins_result,) in zip(pool_addresses, pool_coins_result):
-        if pool_addr in IGNORED_CURVE_POOLS:
+    pool_count = metaregistry.call(node_inquirer=ethereum, method_name='pool_count')
+    all_pools = []
+    for pool_index in range(pool_count):
+        pool_address = to_checksum_address(metaregistry.call(
+            node_inquirer=ethereum,
+            method_name='pool_list',
+            arguments=[pool_index],
+        ))
+        if pool_address in IGNORED_CURVE_POOLS:
             continue
-        pool_coins = []
-        for coin_addr in decoded_coins_result:
-            checksumed_coin_addr = to_checksum_address(coin_addr)
-            if checksumed_coin_addr == ZERO_ADDRESS:
-                # curve returns array of fixed length. So if current pool utilizes fewer
-                # coins than the array's length, first values in the result are the real
-                # coins and all addresses after are just 0x00..00
+        pool_name = metaregistry.call(
+            node_inquirer=ethereum,
+            method_name='get_pool_name',
+            arguments=[pool_address],
+        )
+        gauge_address = to_checksum_address(metaregistry.call(
```

Comment:
In all of this function you can't call `to_checksum_address` you need to do deserialize_evm_address and you need to be checking for error